sdk/core/azure-core/inc/azure/core/operation_status.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+#include <utility> // for std::move
+
+#include "azure/core/strings.hpp"
+
+namespace Azure { namespace Core {
+
+  class OperationStatus {
+    std::string m_value;
+
+  public:
+    // Provide `explicit` conversion from string or types convertible to string:
+    explicit OperationStatus(const std::string& value) : m_value(value) {}
+    explicit OperationStatus(std::string&& value) : m_value(std::move(value)) {}
+    explicit OperationStatus(const char* value) : m_value(value) {}
+
+    // Provide an equality comparison. If the service treats the enumeration case insensitively,
+    // use LocaleInvariantCaseInsensitiveEqual to prevent differing locale settings from affecting
+    // the SDK's behavior:
+    bool operator==(const OperationStatus& other) const noexcept
+    {
+      return Strings::LocaleInvariantCaseInsensitiveEqual(m_value, other.m_value);
+    }
+
+    bool operator!=(const OperationStatus& other) const noexcept
+    {
+      return !(*this == other);
+    }
+
+    // Provide a "Get" accessor
+    const std::string& Get() const noexcept { return m_value; }
+
+    // Provide your example values as static const members
+    const static OperationStatus NotStarted;
+    const static OperationStatus Running;
+    const static OperationStatus Succeeded;
+    const static OperationStatus Cancelled;
+    const static OperationStatus Failed;
```

Comment:
If that is more idiomatic, we need to update the guidelines because they are currently misleading:
https://azure.github.io/azure-sdk/cpp_design.html#enumerations

We also use `const static` here, which we may want to fix (@vhvb1989) :
https://github.com/Azure/azure-sdk-for-cpp/blob/ed1908efa149e2dd8b567c4a36d4dcc56b754769/sdk/core/azure-core/src/http/url.cpp#L79